### PR TITLE
#688: Change order of steps during asked resignation

### PR DESCRIPTION
**Pull request by andreoss**
Per #688:

### CODE REVIEW

**Comment by charlesmike** [comment]
@andreoss thank you for your Pull Request. I'll assign someone to review it soon.

**Comment by charlesmike** [comment]
@amihaiemil please review this Pull Request. Deadline (when it should be merged or closed) is ``2020-11-20T17:40:31.093784``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: self-api/src/main/java/com/selfxdsd/api/pm/Intermediary.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
     /**
      * Next step in line.
      */
-    private Step nextStep;
+    private final Step nextStep;
```
Comment (amihaiemil) [inline self-api/src/main/java/com/selfxdsd/api/pm/Intermediary.java:36]: @andreoss Why did you change this?

File: self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java [hunk 2]
```diff
@@ -28,19 +28,75 @@
...
+        final Conversation ressign = new Resign(
+            next -> {
+                throw new IllegalStateException(
+                    "Should not be called."
+                );
+            }
+        );
+        Mockito.when(event.type()).thenReturn("resign");
+        Mockito.when(event.project()).thenReturn(project);
+        Mockito.when(event.comment()).thenReturn(comment);
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java:71]: @andreoss I think these 3 lines can be removed, these mockings have already been done above (just before instantiating ``resign``.

File: self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java [hunk 3]
```diff
@@ -28,19 +28,75 @@
...
+    public void orderOfStepsIsCorrect() {
+        final Project project = Mockito.mock(Project.class);
+        Mockito.when(project.language()).thenReturn(new English());
+        final Comment comment = Mockito.mock(Comment.class);
+        Mockito.when(comment.author()).thenReturn("mihai");
+        final Event event = Mockito.mock(Event.class);
+        Mockito.when(event.type()).thenReturn("resign");
+        Mockito.when(event.project()).thenReturn(project);
+        Mockito.when(event.comment()).thenReturn(comment);
+        final Conversation ressign = new Resign(
```
Comment (amihaiemil) [inline self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java:62]: @andreoss resign, not ressign

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@andreoss Looks nice, I like the test!

Just a few comments.

File: self-api/src/main/java/com/selfxdsd/api/pm/Intermediary.java (hunk 1 above)
Comment (andreoss) [reply self-api/src/main/java/com/selfxdsd/api/pm/Intermediary.java:36]: @amihaiemil Undone.

File: self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java (hunk 2 above)
Comment (andreoss) [reply self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java]: @amihaiemil Removed it.

File: self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java (hunk 3 above)
Comment (andreoss) [reply self-core-impl/src/test/java/com/selfxdsd/core/managers/ResignTestCase.java]: @amihaiemil Renamed.

**Comment by amihaiemil** [comment]
@rultor merge it please

**Comment by charlesmike** [comment]
@amihaiemil thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).
